shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/queryable/ConvertYamlConfigurationHandler.java

```diff
@@ -0,0 +1,118 @@
+    }
+    
+    private String generateConvertedDistSQL(final YamlProxyDatabaseConfiguration yamlConfig) {
+        StringBuilder convetedDistSQL = new StringBuilder();
+        appendCreateDatabaseDistSQL(yamlConfig.getDatabaseName(), convetedDistSQL);
+        appendAddResourceDistSQL(yamlConfig.getDataSources(), convetedDistSQL);
+        return convetedDistSQL.toString();
+    }
+    
+    private void appendCreateDatabaseDistSQL(final String databaseName, final StringBuilder stringBuilder) {
+        if (databaseName.isEmpty()) {
+            return;
+        }
+        stringBuilder.append(String.format(CREATE_DATABASE, databaseName));
+    }
+    
+    private void appendAddResourceDistSQL(final Map<String, YamlProxyDataSourceConfiguration> databaseResource, final StringBuilder stringBuilder) {
+        if (databaseResource.isEmpty()) {
+            return;
+        }
+        if (null == stringBuilder) {
+            stringBuilder.append(ADD_RESOURCE);
+        } else {
+            stringBuilder.append(String.format(System.lineSeparator()
+                                               + System.lineSeparator()
+                                               + ADD_RESOURCE));
+        }
+        stringBuilder.append(ADD_RESOURCE);
+        for (Map.Entry<String, YamlProxyDataSourceConfiguration> entry : databaseResource.entrySet()) {
+            String resourceName = entry.getKey();
+            String url = entry.getValue().getUrl();
+            String username = entry.getValue().getUsername();
+            String password = entry.getValue().getPassword();
+            String connectionTimeoutMilliseconds = null == entry.getValue().getConnectionTimeoutMilliseconds() ? String.format("")
+                    : String.format("\"connectionTimeoutMilliseconds\"=%s,", entry.getValue().getConnectionTimeoutMilliseconds().toString());
+            String idleTimeoutMilliseconds = null == entry.getValue().getIdleTimeoutMilliseconds() ? String.format("")
+                    : String.format("\"idleTimeoutMilliseconds\"=%s,", entry.getValue().getIdleTimeoutMilliseconds().toString());
+            String maxLifetimeMilliseconds = null == entry.getValue().getMaxLifetimeMilliseconds() ? String.format("")
+                    : String.format("\"getMaxLifetimeMilliseconds\"=%s,", entry.getValue().getMaxLifetimeMilliseconds().toString());
+            String maxPoolSize = null == entry.getValue().getMaxPoolSize() ? String.format("") : String.format("\"maxPoolSize\"=%s,", entry.getValue().getMaxPoolSize().toString());
+            String minPoolSize = null == entry.getValue().getMinPoolSize() ? String.format("") : String.format("\"minPoolSize\"=%s", entry.getValue().getMinPoolSize().toString());
+            stringBuilder.append(String.format(RESOURCES, resourceName, url, username, password, connectionTimeoutMilliseconds,
+                    idleTimeoutMilliseconds, maxLifetimeMilliseconds, maxPoolSize, minPoolSize));
```

Comment:
customPoolProps missed.   (in `YamlProxyDataSourceConfiguration`)

Reply:
Here is a sample for `customPoolProps `:
https://github.com/apache/shardingsphere/issues/17646